lib/enrichment/src/tables.rs

```diff
@@ -49,6 +49,18 @@ pub struct TableRegistry {
     tables: Arc<ArcSwap<Option<TableMap>>>,
 }
 
+/// Pessimistic Eq implementation for caching purposes
+impl PartialEq for TableRegistry {
+    fn eq(&self, other: &Self) -> bool {
+        Arc::ptr_eq(&self.tables, &other.tables) && Arc::ptr_eq(&self.loading, &other.loading)
+            || self.tables.load().is_none()
+                && other.tables.load().is_none()
+                && self.loading.lock().unwrap().is_none()
+                && other.loading.lock().unwrap().is_none()
```

Comment:
We prefer using either `expect` or `unwrap_or_else(|error| panic!(error))` over `unwrap`. This provides both commentary in the code and a proper error message if it ever happens.